sdks/java/core/src/main/java/org/apache/beam/sdk/util/Credentials.java

```diff
@@ -132,15 +135,32 @@ public static Credential getCredential(GcpOptions options)
    * Loads OAuth2 credential from a local file.
    */
   private static Credential getCredentialFromFile(
-      String keyFile, String accountId, Collection<String> scopes)
+          @Nullable String keyFile, @Nullable String accountName, Collection<String> scopes)
       throws IOException, GeneralSecurityException {
-    GoogleCredential credential = new GoogleCredential.Builder()
+
+    if (keyFile == null) {
+      throw new IOException("If accountName is given, also supply a keyFile");
+    }
```

Comment:
Okay, that makes sense.

For visible clarity, I'd prefer if we refactored something like this:

```java
if (keyFile != null) {
   get the credentials;
} else if (account != null) {
   throw IOException("If accountName is given, also supply a keyFile');
}
```

then we can remove the `@Nullable` from the `keyFile` parameter. I think it will make the code generally easier to read and understand.